### PR TITLE
Kademlia exploration

### PR DESCRIPTION
**Pull request by prince-chrismc**
replace #47 

related to #57 (first step)

### CODE REVIEW

**Comment by prince-chrismc** [comment]
I need to test this better but network issue are blocking me

**Comment by prince-chrismc** [comment]
Testing on my home PC in an open network everything works!

File: pyrsia_node/src/docker/error_util.rs [hunk 1]
```diff
@@ -27,6 +28,7 @@ impl fmt::Display for RegistryErrorCode {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         let printable = match &self {
             RegistryErrorCode::BlobUnknown => "BLOB_UNKNOWN".to_string(),
+            RegistryErrorCode::BlobDoesNotExist(hash) => format!("BLOB_DOES_NOT_EXIST({})", hash),
```
Comment (mgrand) [inline pyrsia_node/src/docker/error_util.rs:31]: This is fine. However, there is a crate named strum_macros that generates this for you automatically.  I suggest that you use that and delete this.

**Review by mgrand — APPROVED** [review]

**Review by MitaliBo — APPROVED** [review]
lgtm and i need this code

**Pull request merged**